src/hooks/useLogin.js

```diff
@@ -0,0 +1,38 @@
+import { useAuth } from "~/contexts/AuthContext";
+
+import useAxiosPrivate from "./useAxiosPrivate";
+import config from "../constants/endpoints.json";
+
+const currentConfig = import.meta.env.MODE === "development" ? config.test : config.prod;
+const useLogin = () => {
+  const { setAuthState } = useAuth();
+  const axiosPrivate = useAxiosPrivate();
+  const login = async ({ email, password }) => {
+    const formData = new FormData();
+    formData.append("email", email);
+    formData.append("password", password);
+    const response = await axiosPrivate.post(`/api/${currentConfig.login}`, formData, {
+      headers: {
+        "Content-Type": "multipart/form-data"
```

Comment:
on login why do we need multipart ?

Reply:
> on login why do we need multipart ?
API 's request body   content type  
is multipart

Reply:
> and is it required to have a separate hook for login ?

My thinking process when I did  was  to separate  the logic part from the UI  even though code is not shared an other part of the code 2 ,  I will put in the custom hook  other login related use cases like registration

Reply:
Okay so we can use a more generic name for it then